### PR TITLE
Fix forwarding references and correct creation of std::shared_ptr using std::make_shared

### PR DESCRIPTION
**Pull request by Lezcano**
Changed  the details mentioned in #130.

### CODE REVIEW

File: include/momemta/Pool.h [hunk 1]
```diff
@@ -88,7 +88,7 @@ class Pool {
         friend class Module;
 
         template <typename U>
-        friend class ValueProxy;
+            friend class ValueProxy;
```
Comment (blinkseb) [inline include/momemta/Pool.h]: Please remove indentation

File: include/momemta/Pool.h [hunk 2]
```diff
@@ -137,7 +137,7 @@ class Pool {
         };
 
 
-private:
+    private:
```
Comment (blinkseb) [inline include/momemta/Pool.h]: Please remove indentation

File: include/momemta/impl/Pool.h [hunk 3]
```diff
@@ -91,7 +91,8 @@ template <typename T, typename... Args> std::shared_ptr<T> Pool::put(const Input
 
         // If the block is empty, it's a delayed instantiation. Simply flag the block as valid, and allocate memory for it
         if (it->second.ptr.empty()) {
-            std::shared_ptr<T> ptr(new T(std::forward<Args>(args)...));
+            auto ptr = std::make_shared<T>(std::forward<Args>(args)...);
+
```
Comment (blinkseb) [inline include/momemta/impl/Pool.h]: Unnecessary new line

Comment (Lezcano) [inline include/momemta/impl/Pool.h]: This one is not unnecessary. There are a few advantages that make the second option preferable over the first one. See
https://herbsutter.com/2013/05/29/gotw-89-solution-smart-pointers/

Comment (swertz) [inline include/momemta/impl/Pool.h]: I blinkseb was referring to the empty line 95, not the changed line 94...

Comment (Lezcano) [inline include/momemta/impl/Pool.h]: Oh, true, sorry!

I will correct this things in a minute!

Comment (blinkseb) [inline include/momemta/impl/Pool.h]: Indeed I was referring to the new line, but very interesting link, thanks! :+1:

**Comment by blinkseb** [comment]
All good, thanks a lot and sorry for the delay for the review!

**Pull request merged**